### PR TITLE
Adds a database migration to restore the fileinfos that are deleted

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This migration restores wrongly deleted `FileInfo`s for boards.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53714

### CODE REVIEW

**Review by sbishel — APPROVED** [review]
Much simpler...LGTM

File: server/services/store/sqlstore/migrations/000040_fix_fileinfo_soft_deletes.up.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{{if .plugin}}
+  UPDATE FileInfo
+     SET DeleteAt = 0
+   WHERE CreatorId = 'boards'
+     AND DeleteAt != 0;
```
Comment (lieut-data) [inline server/services/store/sqlstore/migrations/000040_fix_fileinfo_soft_deletes.up.sql:5]: To confirm, does this mean we undelete everything boards has ever deleted? And that we'll figure out what /should/ be deleted in a separate PR?

**Review by lieut-data — APPROVED** [review]

File: server/services/store/sqlstore/migrations/000040_fix_fileinfo_soft_deletes.up.sql (hunk 1 above)
Comment (sbishel) [reply server/services/store/sqlstore/migrations/000040_fix_fileinfo_soft_deletes.up.sql:5]: Correct. Prior to the fix a few weeks ago, no "boards" fileInfos were deleted, so it is essentially moving that fix to here. It was a forward only fix, as it was a low security issue.
